tools/@aws-cdk/cli-args-gen/lib/cli-args-function-gen.ts

```diff
@@ -0,0 +1,97 @@
+import { code, FreeFunction, Module, SelectiveModuleImport, Type, TypeScriptRenderer } from '@cdklabs/typewriter';
+import { EsLintRules } from '@cdklabs/typewriter/lib/eslint-rules';
+import * as prettier from 'prettier';
+import { kebabToCamelCase } from './util';
+import { CliAction, CliConfig } from './yargs-types';
+
+export async function renderCliArgsFunc(config: CliConfig): Promise<string> {
+  const scope = new Module('aws-cdk');
+
+  scope.documentation.push( '-------------------------------------------------------------------------------------------');
+  scope.documentation.push('GENERATED FROM packages/aws-cdk/lib/config.ts.');
+  scope.documentation.push('Do not edit by hand; all changes will be overwritten at build time from the config file.');
+  scope.documentation.push('-------------------------------------------------------------------------------------------');
+
+  scope.addImport(new SelectiveModuleImport(scope, './cli-arguments', ['CliArguments', 'GlobalOptions']));
+  const cliArgType = Type.fromName(scope, 'CliArguments');
+
+  scope.addImport(new SelectiveModuleImport(scope, './settings', ['Command']));
+
+  const createCliArguments = new FreeFunction(scope, {
+    name: 'convertToCliArgs',
+    export: true,
+    returnType: cliArgType,
+    parameters: [
+      { name: 'args', type: Type.ANY },
+    ],
+  });
+  createCliArguments.addBody(code.expr.directCode(buildCliArgsFunction(config)));
+
+  const ts = new TypeScriptRenderer({
+    disabledEsLintRules: [EsLintRules.MAX_LEN], // the default disabled rules result in 'Definition for rule 'prettier/prettier' was not found'
+  }).render(scope);
+
+  return prettier.format(ts, {
+    parser: 'typescript',
+    printWidth: 150,
+    singleQuote: true,
+    quoteProps: 'consistent',
+  });
+}
+
+function buildCliArgsFunction(config: CliConfig): string {
```

Comment:
for all of the following `buildXxx` functions, I am sure there is a more idiomatic way of achieving what I want via typewriter. I have done it this way because it felt the simplest for me as of right now, and I believe it will be easier to refactor later with the benefit of knowing exactly what the generated function should look like.